docs/README.md

```diff
@@ -48,6 +49,12 @@ and infrastructure.
 Please refer to [this document](concepts/ha-and-scaling.md) to understand
 failures scenarios, recovery methods, as well as scaling considerations.
 
+### Resource classes
+
+[Resource classes](concepts/resource-classes.md) filter which resources the
```

Comment:
> If for any reason ingress class is not a good term, let's pick another one, but be strictly consistent about using it everywhere.

That's kind of the thrust of why I chose to use "resource classes": "ingress class" kinda takes its name from the annotation, which was originally applied to Ingress objects only, and then got expanded (at least by us, possibly by other controller vendors) onto other objects, while not really describing what it does.

"Resource class" is also a bit too broad, since it's describing something specific to ingress controllers. "Controller class" maybe fits there, but I don't particularly like it either.

@hbagdi is there anything coming in the future from the service APIs world that would make sense to describe this/top-level resources that a controller will process? Absent that, or if that would be more confusing to use, will probably flip to "ingress class" since, while imperfect, it's the imperfect term among many that at least has the bonus of matching up with the annotation name.

Reply:
eca0d34b0fd939c5593e9c59321ad505648f9d3c assuming the no good alternative case. We can always revert it later.

Reply:
I agree that _ingress class_ is a misnomer because it got expanded, but my request is based on the opinion that a misnomer is better than a misnomer and a synonym present in the docs together.